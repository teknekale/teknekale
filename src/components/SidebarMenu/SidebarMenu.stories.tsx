/* tslint:disable no-var-requires */
const withReadme = require("storybook-readme/with-readme").default;
const SidebarMenuReadme = require("./README.md");

import * as React from "react";
import { storiesOf, action } from "@kadira/storybook";
import { withKnobs, text, boolean } from "@kadira/storybook-addon-knobs";
import SidebarMenu from "./SidebarMenu";

const items = [
  { name: "Home", path: "/", exact: true, icon: "home" },
  { name: "About", path: "/about/", exact: true, icon: "info circle" },
  { name: "Blog", path: "/blog/", exact: false, icon: "newspaper" },
];

const LinkStub = (props: any) =>
  <div {...props} onClick={action(props.to.toString())} >{props.children}</div>;

storiesOf("SidebarMenu", module)
  .addDecorator(withKnobs)
  .addDecorator(withReadme(SidebarMenuReadme))
  .add("default", () => {
    const pathname = text("pathname", "/");

    return (
      <SidebarMenu Link={LinkStub} items={items} pathname={pathname} visible />
    );
  });
