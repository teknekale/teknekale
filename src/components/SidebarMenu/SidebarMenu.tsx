import * as React from "react";
import { MenuProps, MenuItem } from "../Menu";
import { Menu, Icon, Sidebar } from "semantic-ui-react";

interface SidebarMenuProps extends MenuProps {
  visible?: boolean;
}

export default ({ items, pathname, Link, visible }: SidebarMenuProps) => {
  const isActive = (item: MenuItem) => (item.exact) ? pathname === item.path : pathname.startsWith(item.path);
  const activeItem = items.find((item: MenuItem) => isActive(item)) || {} as MenuItem;
  return (
    <Sidebar as={Menu} animation="slide along" width="thin"
      visible={visible} icon="labeled" vertical inverted={activeItem.inverted}>
      {items.map((item) => {
        const active = isActive(item);
        return (
          <Menu.Item as={Link} to={item.path} active={active} key={item.path}>
            <Icon name={item.icon} />
            {item.name}
          </Menu.Item>
        );
      })}
    </Sidebar>
  );
};
