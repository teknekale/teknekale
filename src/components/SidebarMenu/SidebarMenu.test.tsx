import { render } from "enzyme";
import "jest";
import * as React from "react";
import SidebarMenu from "./SidebarMenu";

const items = [
  { name: "Home", path: "/", exact: true },
  { name: "About", path: "/about/", exact: true },
  { name: "Blog", path: "/blog/", exact: false },
];

const LinkStub = (props: any) => <div {...props} />;

describe("SidebarMenu component", () => {
  it("should render correctly", () => {

    const wrapper = render(<SidebarMenu Link={LinkStub} pathname="/" items={items} visible />);
    expect(wrapper).toMatchSnapshot();
  });
});
