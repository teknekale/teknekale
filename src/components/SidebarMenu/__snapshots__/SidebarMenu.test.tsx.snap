// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarMenu component should render correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {
          "class": "ui vertical labeled icon ui slide along left thin visible sidebar menu",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "active item",
              "to": "/",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "aria-hidden": "true",
                  "class": "icon",
                },
                "children": Array [],
                "name": "i",
                "next": Object {
                  "data": "Home",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "text",
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
              },
              Object {
                "data": "Home",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "aria-hidden": "true",
                    "class": "icon",
                  },
                  "children": Array [],
                  "name": "i",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                },
                "type": "text",
              },
            ],
            "name": "div",
            "next": Object {
              "attribs": Object {
                "class": "item",
                "to": "/about/",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "aria-hidden": "true",
                    "class": "icon",
                  },
                  "children": Array [],
                  "name": "i",
                  "next": Object {
                    "data": "About",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "text",
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                },
                Object {
                  "data": "About",
                  "next": null,
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "aria-hidden": "true",
                      "class": "icon",
                    },
                    "children": Array [],
                    "name": "i",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                  },
                  "type": "text",
                },
              ],
              "name": "div",
              "next": Object {
                "attribs": Object {
                  "class": "item",
                  "to": "/blog/",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "aria-hidden": "true",
                      "class": "icon",
                    },
                    "children": Array [],
                    "name": "i",
                    "next": Object {
                      "data": "Blog",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "text",
                    },
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                  },
                  Object {
                    "data": "Blog",
                    "next": null,
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {
                        "aria-hidden": "true",
                        "class": "icon",
                      },
                      "children": Array [],
                      "name": "i",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                    },
                    "type": "text",
                  },
                ],
                "name": "div",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
          },
          Object {
            "attribs": Object {
              "class": "item",
              "to": "/about/",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "aria-hidden": "true",
                  "class": "icon",
                },
                "children": Array [],
                "name": "i",
                "next": Object {
                  "data": "About",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "text",
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
              },
              Object {
                "data": "About",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "aria-hidden": "true",
                    "class": "icon",
                  },
                  "children": Array [],
                  "name": "i",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                },
                "type": "text",
              },
            ],
            "name": "div",
            "next": Object {
              "attribs": Object {
                "class": "item",
                "to": "/blog/",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "aria-hidden": "true",
                    "class": "icon",
                  },
                  "children": Array [],
                  "name": "i",
                  "next": Object {
                    "data": "Blog",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "text",
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                },
                Object {
                  "data": "Blog",
                  "next": null,
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "aria-hidden": "true",
                      "class": "icon",
                    },
                    "children": Array [],
                    "name": "i",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                  },
                  "type": "text",
                },
              ],
              "name": "div",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
            },
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "active item",
                "to": "/",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "aria-hidden": "true",
                    "class": "icon",
                  },
                  "children": Array [],
                  "name": "i",
                  "next": Object {
                    "data": "Home",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "text",
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                },
                Object {
                  "data": "Home",
                  "next": null,
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "aria-hidden": "true",
                      "class": "icon",
                    },
                    "children": Array [],
                    "name": "i",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                  },
                  "type": "text",
                },
              ],
              "name": "div",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
            },
            "type": "tag",
          },
          Object {
            "attribs": Object {
              "class": "item",
              "to": "/blog/",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "aria-hidden": "true",
                  "class": "icon",
                },
                "children": Array [],
                "name": "i",
                "next": Object {
                  "data": "Blog",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "text",
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
              },
              Object {
                "data": "Blog",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "aria-hidden": "true",
                    "class": "icon",
                  },
                  "children": Array [],
                  "name": "i",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                },
                "type": "text",
              },
            ],
            "name": "div",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "item",
                "to": "/about/",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "aria-hidden": "true",
                    "class": "icon",
                  },
                  "children": Array [],
                  "name": "i",
                  "next": Object {
                    "data": "About",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "text",
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                },
                Object {
                  "data": "About",
                  "next": null,
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "aria-hidden": "true",
                      "class": "icon",
                    },
                    "children": Array [],
                    "name": "i",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                  },
                  "type": "text",
                },
              ],
              "name": "div",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "active item",
                  "to": "/",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "aria-hidden": "true",
                      "class": "icon",
                    },
                    "children": Array [],
                    "name": "i",
                    "next": Object {
                      "data": "Home",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "text",
                    },
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                  },
                  Object {
                    "data": "Home",
                    "next": null,
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {
                        "aria-hidden": "true",
                        "class": "icon",
                      },
                      "children": Array [],
                      "name": "i",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                    },
                    "type": "text",
                  },
                ],
                "name": "div",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
              },
              "type": "tag",
            },
            "type": "tag",
          },
        ],
        "name": "div",
        "next": null,
        "parent": null,
        "prev": null,
        "root": [Circular],
        "type": "tag",
      },
    ],
    "name": "root",
    "next": null,
    "parent": null,
    "prev": null,
    "type": "root",
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xmlMode": false,
  },
}
`;
